### PR TITLE
Users: Fix org user always getting org id = 1 on auto assign false

### LINKED ISSUE #63562: Regression 9.4.0-beta1; allow_org_create creates org, but doesn't add user to it. Adds user as Admin on orgID=1 instead.
**What happened**:
OSS Edition, configured to auto create organizations and NOT add user to `orgID=1`.

Organization is created when user is registering, either with GitHub or GMail or "Sign up" buttons. User is NOT added to the organization that is created, and instead added to OrgId=1

**What you expected to happen**:
That the created user is NOT added to the  `orgID=1` and that the user IS ADDED to the organization that the user just created.

**How to reproduce it (as minimally and precisely as possible)**:
```
sudo dpkg --purge grafana
sudo rm -rf /var/log/grafana/ /var/lib/grafana/ /etc/grafana/

wget https://dl.grafana.com/oss/release/grafana_9.4.0~beta1_amd64.deb
sudo dpkg --install grafana_9.4.0~beta1_amd64.deb 
sudo vi /etc/grafana/grafana.ini
```
change to;
```
allow_sign_up = true
allow_org_create = true
auto_assign_org = false
auto_assign_org_id = 1
auto_assign_org_role = Viewer
```
and then
```
sudo /bin/systemctl daemon-reload
sudo /bin/systemctl enable grafana-server
sudo systemctl start grafana-server
```

WAIT for grafana to complete first start up. `tail -f /var/log/grafana/grafana.log`

Then on start page, do "Sign up". I get to ?orgId=1 on 9.4.0-beta1 but to ?orgId=3 on 9.3.6

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.4.0-beta1
- Data source type & version: Irrelevant
- OS Grafana is installed on: Linux Debian
- User OS & Browser: Shouldn't matter
- Grafana plugins: Irrelevant
- Others:

### PR DESCRIPTION
**Pull request by Jguer**
**What is this feature?**

Fix org user always getting org id = 1 on auto assign false.

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/63562

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/org/orgimpl/store.go [hunk 1]
```diff
@@ -76,9 +76,12 @@ func (ss *sqlStore) Insert(ctx context.Context, org *org.Org) (int64, error) {
 	var orgID int64
 	var err error
 	err = ss.db.WithDbSession(ctx, func(sess *db.Session) error {
-		if orgID, err = sess.InsertOne(org); err != nil {
+		if _, err = sess.Insert(org); err != nil {
 			return err
 		}
+
+		orgID = org.ID
```
Comment (eleijonmarck) [inline pkg/services/org/orgimpl/store.go:83]: do we want to have checks here instead of the org.ID != 0 down below?

Comment (Jguer) [reply pkg/services/org/orgimpl/store.go:83]: trying to minimize flow changes for this fix due to backport and undefined behavior.

For 9.5 we can start looking at shoring up the tests for this and adding more guards

**Review by eleijonmarck — APPROVED** [review]
LGTM, did not test locally, just pulled and check through code

**Review by mgyongyosi — APPROVED** [review]

**Comment by vtorosyan** [comment]
https://github.com/grafana/grafana/issues/63562 related issue

**Pull request merged**